### PR TITLE
feature(mycollection filter): adding artist name filter

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Feature**

This PR resolves []

### Description

This also fixes a small bug with the uniqBy in the medium. we uniqBy value, not name, so we are sure they are different.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add artist name filter in mycollection - pavlos

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (add artist name filter in mycollection - pavlos)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 099cda146f1f8b16489fe87f6ec7e53ae3b71e0e
</p>

**Review by olerichter00 — APPROVED** [review]
LGTM

File: src/lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid.tsx [hunk 1]
```diff
@@ -469,6 +469,10 @@ export const InfiniteScrollMyCollectionArtworksGridContainer = createFragmentCon
             }
             artistNames
             medium
+            artist {
+              internalID
+              name
+            }
```
Comment (MounirDhahri) [inline src/lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid.tsx:475]: **nitpick:** is this used somewhere? if it's only needed for my collection using the skip or include directives as in line 477

**Review by MounirDhahri — COMMENTED** [review]
nice, I only left one small comment. LGTM though

**Pull request merged**